src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -76,7 +75,22 @@ const submitEncryptModeForm: RequestHandler = async (req, res) => {
     const { errorMessage, statusCode } = mapRouteError(formResult.error)
     return res.status(statusCode).json({ message: errorMessage })
   }
-  const form = formResult.value
+
+  const checkFormIsEncryptModeResult = checkFormIsEncryptMode(formResult.value)
+  if (checkFormIsEncryptModeResult.isErr()) {
+    logger.error({
+      message:
+        'Trying to submit non-encrypt mode submission on encrypt-form submission endpoint',
+      meta: logMeta,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      checkFormIsEncryptModeResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+  const form = checkFormIsEncryptModeResult.value
```

Comment:
actually i think neverthrow solves this problem of repeated error checking - we can just chain this into the next method and be confident that the form is already encrypted!

then, the `mapErr` will be applied to the result of the whole expression (inclusive) of this! we could do a pass through on the err (using `mapErr`) so that the type is preserved but logs are still present too. 

idk if it's beyond the scope of this PR but yea, just something to think about, especially because lines 70 - 186 is essentially just checking the form

Reply:
thanks for the review! yup the end goal will definitely to refactor the whole pipeline to use the same map/mapErr pattern as the email-submission endpoint. but that should be left for later after all the upcoming work on the encrypt-submission pipeline is completed